Modules/_decimal/_decimal.c

```diff
@@ -3672,34 +3673,45 @@ dec_as_long(PyObject *dec, PyObject *context, int round)
     }
 
     status = 0;
-    ob_digit = NULL;
+
+    int64_t val = mpd_qget_i64(x, &status);
+    if (!status) {
+        mpd_del(x);
+        return PyLong_FromInt64(val);
+    }
+
+    n = (mpd_sizeinbase(x, 2) +
+         layout->bits_per_digit - 1) / layout->bits_per_digit;
```

Comment:
```suggestion
    const uint8_t bpd = layout->bits_per_digit;
    n = (mpd_sizeinbase(x, 2) + bpd - 1) / bpd;
```

The compiler should be smart enough to inline this (hopefully).